### DISCUSSION TITLE
How to Include an Access Token in the Headers Using httpSubscriptionLink in a Web Environment

### DISCUSSION THREAD

**Discussion by tomox99** [trpc/trpc#6071] [Q&A] [answered]
**Description:**
Hello, I'm implementing Server-Sent Events (SSE) using the httpSubscriptionLink as per the [tRPC documentation](https://trpc.io/docs/client/links/httpSubscriptionLink).

I need to include an access token in the headers in a web environment. I'm using httpSubscriptionLink on the client-side.

**Question:**
- The documentation introduces a way to include an access token in the headers, which is recommended for non-web environments, but how should this be implemented in a web environment?

**Additional Info**

- I was able to include an access token in the headers and send a request to an external API using the method recommended for non-web environments, as described in the documentation.

**Setup:**
- Frontend: Next.js with the vanilla tRPC client
- Backend: Express using the tRPC Express adapter

Any help would be greatly appreciated!

**Comment by oleasteo** [comment, marked as answer]
tRPC uses `EventSource` for the `httpSubscriptionLink`. `EventSource` [does not support custom HTTP headers](https://github.com/whatwg/html/issues/2177), unfortunately. Thus, the Ponyfill as described in the docs is the only way to achieve this presently (it uses `fetch` under the hood).

The recommended way to authenticate http subscriptions in a web environment are cookies instead.

**Reply by KATT** [reply]
Thanks, this is the way.

Ideally, in a web application you use cookies so no ponyfill is needed.

cc @tsux-ssbu, please revisit the documentation on subscriptions and this link as we revamped it recently

**Reply by tomox99** [reply]
Thank you for your reply. I will check the documents. :)
